### PR TITLE
Adjustment of code generator to address #1368

### LINKED ISSUE #1368: Memory error with reactions writing to multiple ports
The following program fails with either `FATAL ERROR: Out of memory!` or a segmentation fault (depending on insertion of printfs):
```
target C;
reactor C {
    output x:int;
    output z:int;
    timer t(0, 100 msec);
    reaction(t) -> x, z {=
        lf_set(x, 42);
        lf_set(z, 44);
    =}
}
main reactor {
    c = new C();
    reaction(c.z) {=
        lf_print("c.z = %d", c.z->value);
    =}
}
```
image

### PR DESCRIPTION
**Pull request by lsk567**
Fixes #1368.

### CODE REVIEW

File: test/C/src/MultipleOutputs.lf [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        lf_set(x, 42);
+        lf_set(z, 44);
+    =}
+}
+main reactor {
+    c = new C();
+    reaction(c.z) {=
+        lf_print("c.z = %d", c.z->value);
+    =}
+}
```
Comment (lhstrh) [inline test/C/src/MultipleOutputs.lf:22]:
```suggestion
}

```

**Review by lhstrh — APPROVED** [review]
Awesome, thanks for tracking this down! You might want to run the formatter on the test to make sure it is formatted correctly and avoid a test failure.

**Comment by lhstrh** [comment]
Note: I ran `./gradlew runLff --args "test/C/src/MultipleOutputs.lf"`

**Review by edwardalee — APPROVED** [review]
Great catch!  To be a test, the MultiportOutputs.lf needs to call `lf_print_error_and_exit` if the reaction to to `c.z` is not invoked. That can be handled in a shutdown reaction that checks a state variable.

**Pull request merged**